src/modules/external/aha/components/AhaScore.tsx

```diff
@@ -39,7 +42,14 @@ const AhaScore = ({
 
   const [errorState, setErrorState] = useState<ErrorState>(emptyErrorState);
 
-  const [hasScore, setHasScore] = useState<boolean>(false);
+  // Use RHF to watch the current value of the score field.
+  // This is to disable the fetch button after unlocking an assessment that's already fetched the score.
+  const values = useDynamicFieldWatchValues([SCORE_LINK_ID]);
+  const existingValue = values[SCORE_LINK_ID];
+
+  const [hasScore, setHasScore] = useState<boolean>(
+    !!existingValue && existingValue >= 0
+  );
```

Comment:
Since we're _watching_ the SCORE_LINK_ID field now, do we still need to maintain an internal state field for whether the score exists or not?
I haven't tested this out since I don't have the full setup locally, but I wonder if it behaves the same way if `hasScore` was populated from the watched form field:

```suggestion
  // Use RHF to watch the current value of the score field.
  // This is to disable the fetch button after unlocking an assessment that's already fetched the score.
  const values = useDynamicFieldWatchValues([SCORE_LINK_ID]);
  const existingValue = values[SCORE_LINK_ID];

  const hasScore = !!existingValue && existingValue >= 0; // this should update whenever existingValue changes, without needing to call setHasScore all over
```
If that works the same way, it's less stuff to manage in this component, which is nice. I might be missing some reason for why this doesn't work though.

Reply:
Thanks, yeah this seems like a nice improvement! It seems to work well with a minor modification to accommodate 0 as a valid value for the score